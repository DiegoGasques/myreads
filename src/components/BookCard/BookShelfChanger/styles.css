.book-shelf-changer-2 {
  width: 30px;
  height: 30px;
  margin: 10px 0;
  border-radius: 50%;
  background: #4a4063;
  background-image: url("../../../icons/arrow-drop-down.svg");
  background-repeat: no-repeat;
  background-position: center;
  background-size: 20px;
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
}

.book-shelf-changer-2 select {
  width: 100%;
  height: 100%;
  opacity: 0;
  cursor: pointer;
}
