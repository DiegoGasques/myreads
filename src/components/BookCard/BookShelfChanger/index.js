import React from "react";
import PropTypes from "prop-types";

import "./styles.css";

function BookShelfChanger({ handleUpdate, currValue }) {
  return (
    <div className="book-shelf-changer-2">
      <select value={currValue} onChange={e => handleUpdate(e.target.value)}>
        <option value="move" disabled>
          Move to...
        </option>
        <option value="currentlyReading">Currently Reading</option>
        <option value="wantToRead">Want to Read</option>
        <option value="read">Read</option>
        <option value="none">None</option>
      </select>
    </div>
  );
}

BookShelfChanger.propTypes = {
  handleUpdate: PropTypes.func.isRequired
};

export default BookShelfChanger;
