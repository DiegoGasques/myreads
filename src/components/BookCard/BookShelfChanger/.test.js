import React from "react";
import { shallow } from "enzyme";
import BookShelfChanger from "./index";
import toJSON from "enzyme-to-json";
import { books } from "../../../testFixtures";

it("Should match the component snapshot", () => {
    const handleUpdate = jest.fn();
  const wrapper = shallow(<BookShelfChanger currValue="wantToRead" handleUpdate={handleUpdate} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("Should call handleUpdate on change event with right value", () => {
    const handleUpdateSpy = jest.fn();
    const value = 'currentlyReading';
    const wrapper = shallow(<BookShelfChanger currValue="wantToRead" handleUpdate={handleUpdateSpy} />);
    wrapper.find('select').simulate('change', {
        target: {
            value
        }
    })
    expect(handleUpdateSpy).toHaveBeenLastCalledWith(value);
});