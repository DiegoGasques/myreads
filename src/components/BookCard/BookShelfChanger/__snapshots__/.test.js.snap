// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the component snapshot 1`] = `
<div
  className="book-shelf-changer-2"
>
  <select
    onChange={[Function]}
    value="wantToRead"
  >
    <option
      disabled={true}
      value="move"
    >
      Move to...
    </option>
    <option
      value="currentlyReading"
    >
      Currently Reading
    </option>
    <option
      value="wantToRead"
    >
      Want to Read
    </option>
    <option
      value="read"
    >
      Read
    </option>
    <option
      value="none"
    >
      None
    </option>
  </select>
</div>
`;
